src/containers/Doctor.js

```diff
@@ -0,0 +1,82 @@
+import { useLocation } from 'react-router-dom';
+import { useState } from 'react';
+import { useSelector } from 'react-redux';
+import userAuth from '../helpers/userAuth';
+import { useHttp } from '../hooks/http';
+import { loadDoctor } from '../actions';
+import AppointmentModal from './AppointmentModal';
+
+const Doctor = () => {
+  const location = useLocation();
+
+  const loggedIn = useSelector((state) => state.user);
+  const doctor = useSelector((state) => state.doctor);
+
+  const [doc, setDoc] = useState({
```

Comment:
This doesn't seem the proper use of React's useState.

React included this feature/hook in order for us to have a more efficient way to manage local state changes. For instance, we should split the local state into:

const [id, setId] = useState(0);
const [name, setName] = useState('');
etc